include/maths/CKdTree.h

```diff
@@ -181,6 +181,7 @@ class CKdTree {
     //! \note The \p points are moved into place.
     void build(TPointVec&& points) {
         this->build(points.begin(), points.end(), True());
+        points.clear();
```

Comment:
I think you uncovered a bug in the usage of rvalues, `points`is not really moved until you call `std::move()` on it before usage, points is still a reference. That's why you had to add `clear()` to actually free memory. In code:

```
void build(TPointVec&& points) {
        TPointVec movedPoints = std::move(points);
        this->build(movedPoints.begin(), movedPoints.end(), True());
}
```

should be the way to do it, keeping the existing interface, but I think the code would be easier to understand if you change points to a real reference `TPointVec&` and call `clear()` on it like in this fix.

Note: Maybe better to fix in a separate PR?

Reply:
Ok I agree, this is really superfluous to this PR and is aimed at transient memory usage in outlier detection code. I have a PR (upcoming) which addresses some other inefficiencies and I need to add memory instrumentation. I'll back out this one line and leave for later.